servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -334,100 +319,19 @@ public StreamingHttpRequest newRequest(final HttpRequestMethod method, final Str
-            final String loggerName, final LogLevel logLevel, final BooleanSupplier logUserData) {
-        builderTemplate.enableWireLogging(loggerName, logLevel, logUserData);
-        return this;
-    }
-
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> protocols(HttpProtocolConfig... protocols) {
-        builderTemplate.protocols(protocols);
-        return this;
-    }
-
     @Override
     public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> initializer(
             final SingleAddressInitializer<HostAndPort, InetSocketAddress> initializer) {
         this.singleAddressInitializer = requireNonNull(initializer);
         return this;
     }
 
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> appendConnectionFilter(
-            final StreamingHttpConnectionFilterFactory factory) {
-        builderTemplate.appendConnectionFilter(factory);
-        return this;
-    }
-
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> appendConnectionFactoryFilter(
-            final ConnectionFactoryFilter<InetSocketAddress, FilterableStreamingHttpConnection> factory) {
-        builderTemplate.appendConnectionFactoryFilter(factory);
-        return this;
-    }
-
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> appendClientFilter(
-            final StreamingHttpClientFilterFactory function) {
-        clientFilterFactory = appendClientFilter(clientFilterFactory, function.asMultiAddressClientFilter());
-        return this;
-    }
-
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> disableHostHeaderFallback() {
-        builderTemplate.disableHostHeaderFallback();
-        return this;
-    }
-
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> allowDropResponseTrailers(
-            final boolean allowDrop) {
-        builderTemplate.allowDropResponseTrailers(allowDrop);
-        return this;
-    }
-
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> autoRetryStrategy(
-            final AutoRetryStrategyProvider autoRetryStrategyProvider) {
-        builderTemplate.autoRetryStrategy(autoRetryStrategyProvider);
-        return this;
-    }
-
     @Override
     public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> maxRedirects(final int maxRedirects) {
         this.maxRedirects = maxRedirects;
         return this;
     }
 
-    @Override
-    public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> unresolvedAddressToHost(
-            Function<HostAndPort, CharSequence> unresolvedAddressToHostFunction) {
-        this.unresolvedAddressToHostFunction = requireNonNull(unresolvedAddressToHostFunction);
-        return this;
-    }
-
     @Override
     public MultiAddressHttpClientBuilder<HostAndPort, InetSocketAddress> appendClientFilter(
```

Comment:
I am wondering whether this method should now become deprecated as this behaviour can be achieved using the `SingleAddressInitializer` and the code can be simplified.

Reply:
If you have questions then no doubt others do as well. I prefer one mechanism especially when there is ambiguity about differences in behavior between multiple options.

Reply:
What I can suggest now:
1. create a separate PR: add `@Deprecated` annotation, leave existing logic, release soon and let users digest the change
1. merge this PR in a breaking-change release
1. another PR, another release removes the recently `@Deprecated`  method and cleans up the logic

Points 2. and 3. can be merged if we anyways have to wait until the deprecation is released.

Reply:
+1, those also have to be deprecated, including `MultiAddressHttpClientFilterFactory` interface. We can cherry-pick (1) into 0.41 branch and do (2) and (3) together in this PR.